cdap-ui/app/cdap/services/datasource/index.js

```diff
@@ -230,26 +240,33 @@ export default class Datasource {
       type: 'POLL',
     };
 
-    this.socketSend('poll-start', generatedResource);
+    this.socketSend('request', generatedResource);
 
     return observable;
   }
 
+  startClientPoll = (resourceId, interval) => {
+    const intervalTimer = setInterval(() => {
```

Comment:
There might be a potential issue with using interval. We are queing request every 10 seconds, so if for some reason response takes longer than 10 seconds, the request will keep piling. Would it be better to do this as a timeout on the response callback instead?

Reply:
Good point. Have updated the PR